libs/features/personalization/personalization.js

```diff
@@ -271,30 +269,87 @@ function parsePlaceholders(placeholders, config, selectedVariantName = '') {
+    }
+    return entitlements;
+  });
+})();
+
+// remove underscores from entitlement names
+const rmUs = (obj) => Object.entries(obj).reduce((newObj, [key, val]) => {
+  newObj[key.replaceAll('_', '')] = val;
+  return newObj;
+}, {});
+
+const getFlatEntitlements = async () => {
+  const ents = await loadEntitlements();
+  return { ...rmUs(ents.arrangment_codes), ...rmUs(ents.clouds), ...rmUs(ents.fulfilled_codes) };
+};
+
+const checkForParamMatch = (paramStr) => {
+  const [name, val] = paramStr.split('param-')[1].split('=');
+  if (!name) return false;
+  const searchParamVal = PAGE_URL.searchParams.get(name);
+  if (searchParamVal !== null) {
+    if (val) return val === searchParamVal;
+    return true; // if no val is set, just check for existence of param
+  }
+  return false;
+};
+
+const checkForEntitlementMatch = (name, entitlements) => {
+  const entName = name.split('ent-')[1];
+  if (!entName) return false;
+  return entitlements[entName];
+};
+
+async function getPersonalizationVariant(manifestPath, variantNames = [], variantLabel = null) {
   const config = getConfig();
-  let manifestFound = false;
   if (config.mep?.override !== '') {
-    config.mep?.override.split(',').forEach((item) => {
+    let manifest;
+    config.mep?.override.split(',').some((item) => {
       const pair = item.trim().split('--');
       if (pair[0] === manifestPath && pair.length > 1) {
-        // eslint-disable-next-line prefer-destructuring
-        manifestFound = pair[1];
+        [, manifest] = pair;
+        return true;
       }
+      return false;
     });
-    if (manifestFound) return manifestFound;
+    if (manifest) return manifest;
   }
 
-  const tagNames = Object.keys(PERSONALIZATION_TAGS);
-  const matchingVariant = variantNames.find((variant) => {
-    // handle multiple variants that are space / comma delimited
-    const names = variant.split(',').map((v) => v.trim()).filter(Boolean);
-    return names.some((name) => {
-      if (name === variantLabel) return true;
-      if (name.startsWith('param-')) return checkForParamMatch(name);
-      return tagNames.includes(name) && PERSONALIZATION_TAGS[name]();
-    });
-  });
+  const personalizationTags = Object.keys(PERSONALIZATION_TAGS);
+  const entitlementTags = Object.keys(ENTITLEMENT_TAGS);
+  const variantInfo = variantNames.reduce((acc, name) => {
+    const vNames = name.split(',').map((v) => v.trim()).filter(Boolean);
+    acc[name] = vNames;
+    acc.allNames = [...acc.allNames, ...vNames];
+    return acc;
+  }, { allNames: [] });
+
+  const hasEntitlementPrefix = variantInfo.allNames.some((name) => name.startsWith('ent-'));
+  const hasEntitlementTag = entitlementTags.some((tag) => variantInfo.allNames.includes(tag));
+
+  let entitlements = [];
+  if (hasEntitlementPrefix || hasEntitlementTag) {
+    entitlements = await getFlatEntitlements();
+  }
+
+  const matchingVariant = variantNames.find((variant) => variantInfo[variant].some((name) => {
```

Comment:
This is a bit hard to parse. You could break out the function passed to `some` to its own named function to increase readability.